source/docs/software/advanced-control/trajectories/troubleshooting.rst

```diff
@@ -0,0 +1,241 @@
+        m_odometry.update(Rotation2d.fromDegrees(getHeading()), m_leftEncoder.getDistance(),
+            m_rightEncoder.getDistance());
+
+        var translation = m_odometry.getPoseMeters().getTranslation();
+        m_xEntry.setNumber(translation.getX());
+        m_yEntry.setNumber(translation.getY());
+    }
+
+   .. code-tab:: c++
+
+    NetworkTableEntry m_xEntry = nt::NetworkTableInstance::GetDefault().GetTable("troubleshooting")->GetEntry("X");
+    NetworkTableEntry m_yEntry = nt::NetworkTableInstance::GetDefault().GetTable("troubleshooting")->GetEntry("Y");
+
+    void DriveSubsystem::Periodic() {
+        // Implementation of subsystem periodic method goes here.
+        m_odometry.Update(frc::Rotation2d(units::degree_t(GetHeading())),
+                            units::meter_t(m_leftEncoder.GetDistance()),
+                            units::meter_t(m_rightEncoder.GetDistance()));
+
+        auto translation = m_odometry.GetPose().Translation();
+        m_xEntry.SetDouble(translation.X().to<double>());
+        m_yEntry.SetDouble(translation.Y().to<double>());
+    }
+
+2. Lay out a tape measure parallel to your robot and push your robot out about one meter along the tape measure. Lay out a tape measure along the Y axis and start over, pushing your robot one meter along the X axis and one meter along the Y axis in a rough arc.
+3. Compare X and Y reported by the robot to actual X and Y. If X is off by more than 5 centimeters in the first test then you should check that you measured your wheel diameter correctly, and that your wheels are not warn down. If the second test is off by more than 5 centimeters in either X or Y then your track width (distance from the center of the left wheel to the center of the right wheel) may be incorrect; if you're sure that you measured the track width correctly with a tape measure then your robot's wheels may be slipping in a way that is not accounted for by track width--if this is the case then you should :ref:`run the track width characterization <docs/software/wpilib-tools/robot-characterization/characterization-routine:Running Tests>` and use that track width instead of the one from your tape measure.
+
+Verify Feedforward
+^^^^^^^^^^^^^^^^^^
+If your feedforwards are bad then the P controllers for each side of the robot will not track as well, and your ``DifferentialDriveVoltageConstraint`` will not limit your robot's acceleration accurately. We mostly want to turn off the wheel P controllers so that we can isolate and test the feedforwards.
+
+1. First, we must set disable the P controller for each wheel. Set the ``P`` gain to 0 for every controller. In the ``RamseteCommand`` example, you would set ``kPDriveVel`` to 0:
+
+.. tabs::
+
+  .. group-tab:: Java
+
+    .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/master/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/ramsetecommand/RobotContainer.java
```

Comment:
Should this reference a commit so that it does not change?
I've noticed in a few other places the examples change and it breaks the line numbers.